src/main/java/org/broadinstitute/consent/http/resources/DataRequestVoteResource.java

```diff
@@ -253,18 +258,20 @@ private void createDataOwnerElection(Vote vote, DataAccessRequest dar) throws Me
             finalVote = CollectionUtils.isNotEmpty(finalVotes) ? finalVotes.get(0) : null;
             agreementVote = vote;
         }
-        if((finalVote != null && finalVote.getVote() != null && finalVote.getVote()) && (agreementVote == null || (agreementVote != null && agreementVote.getVote() != null))){
-            List<Dataset> needsApprovedDataSets = datasetService.findNeedsApprovalDataSetByObjectId(dar.getDatasetIds());
+        if((finalVote != null && finalVote.getVote() != null && finalVote.getVote()) && (agreementVote == null || agreementVote.getVote() != null)){
+            DarCollection collection = darCollectionService.getByReferenceId(referenceId);
+            List<Integer> datasetIds = Objects.nonNull(collection) ? collection.getDars().get(referenceId).getDatasetIds() : List.of();
```

Comment:
This should work, we're pulling the dataset ids in the `findDARCollectionByReferenceId` dao call